pkg/pillar/cmd/msrv/msrv.go

```diff
@@ -0,0 +1,642 @@
+		select {
+		case change := <-msrv.subGlobalConfig.MsgChan():
+			msrv.subGlobalConfig.ProcessChange(change)
+		case <-stillRunning.C:
+		}
+		msrv.PubSub.StillRunning(agentName, warningTime, errorTime)
+	}
+	msrv.Log.Noticef("Processed GlobalConfig")
+
+	// Publish network metrics (interface counters, etc.)
+	interval := time.Duration(msrv.metricInterval) * time.Second
+	max := float64(interval) / publishTickerDivider
+	min := max * 0.3
+	publishTicker := flextimer.NewRangeTicker(time.Duration(min),
+		time.Duration(max))
+	msrv.publishTicker = &publishTicker
+
+	if err = msrv.Activate(); err != nil {
+		return err
+	}
+
+	for {
+		select {
+
+		case change := <-msrv.subNetworkInstanceStatus.MsgChan():
+			msrv.subNetworkInstanceStatus.ProcessChange(change)
+
+		case change := <-msrv.subEdgeNodeInfo.MsgChan():
+			msrv.subEdgeNodeInfo.ProcessChange(change)
+
+		case change := <-msrv.subControllerCert.MsgChan():
+			msrv.subControllerCert.ProcessChange(change)
+
+		case change := <-msrv.subEdgeNodeCert.MsgChan():
+			msrv.subEdgeNodeCert.ProcessChange(change)
+
+		case change := <-msrv.subAppNetworkConfig.MsgChan():
+			msrv.subAppNetworkConfig.ProcessChange(change)
+
+		case change := <-msrv.subAppNetworkConfigAg.MsgChan():
+			msrv.subAppNetworkConfigAg.ProcessChange(change)
+
+		case change := <-msrv.subLocationInfo.MsgChan():
+			msrv.subLocationInfo.ProcessChange(change)
+
+		case change := <-msrv.subWwanStatus.MsgChan():
+			msrv.subWwanStatus.ProcessChange(change)
+
+		case change := <-msrv.subWwanMetrics.MsgChan():
+			msrv.subWwanMetrics.ProcessChange(change)
+
+		case change := <-msrv.subDomainStatus.MsgChan():
+			msrv.subDomainStatus.ProcessChange(change)
+
+		case change := <-msrv.subAppNetworkStatus.MsgChan():
+			msrv.subAppNetworkStatus.ProcessChange(change)
+
+		case change := <-msrv.subAppInstanceStatus.MsgChan():
+			msrv.subAppInstanceStatus.ProcessChange(change)
+
+		case change := <-msrv.subPatchEnvelopeInfo.MsgChan():
+			msrv.subPatchEnvelopeInfo.ProcessChange(change)
+
+		case change := <-msrv.subVolumeStatus.MsgChan():
+			msrv.subVolumeStatus.ProcessChange(change)
+
+		case change := <-msrv.subContentTreeStatus.MsgChan():
+			msrv.subContentTreeStatus.ProcessChange(change)
+
+		case <-ctx.Done():
+			return nil
+
+		case <-msrv.publishTicker.C:
+			start := time.Now()
+			msrv.Log.Traceln("publishTicker at", time.Now())
+
+			msrv.PublishPatchEnvelopesUsage()
+
+			msrv.PubSub.CheckMaxTimeTopic(agentName, "publishMetrics", start,
+				warningTime, errorTime)
+
+		case <-stillRunning.C:
+		}
+		msrv.PubSub.StillRunning(agentName, warningTime, errorTime)
+	}
+}
+
+// MakeMetadataHandler creates http.Handler to be used by LinuxNIReconciler
+func (msrv *Msrv) MakeMetadataHandler() http.Handler {
+	r := chi.NewRouter()
+	zedcloudCtx := zedcloud.NewContext(msrv.Log, zedcloud.ContextOptions{})
+
+	r.Route("/eve/v1", func(r chi.Router) {
+		r.Get("/network.json", handleNetwork(msrv))
```

Comment:
These handlers can also be methods of Msrv, and here you can pass `msrv.handleNetwork` etc.

Reply:
done